fix: UTC-325: By Media Start Time sort doesn't work with Video Object Tracking template

nass600 opened the pull request:
Extended the allow list of tag combinations to trigger the sorting via `start media time` and handled the `videorectangleregion` data to sort accordingly

After:

https://github.com/user-attachments/assets/2e5bdea1-5925-4b57-8193-b87f64a0ccc8



<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nass600 committed:
fix: UTC-325: By Media Start Time sort doesn't work with Video Object Tracking template

hlomzik commented on web/libs/editor/src/stores/RegionStore.js:
```diff
@@ -280,6 +280,15 @@ export default types
+        if (region.type === "videorectangleregion" && region.sequence && region.sequence.length > 0) {
+          // Get the first keyframe's frame number
+          const firstKeyframe = region.sequence[0];
+          if (firstKeyframe && typeof firstKeyframe.frame === "number") {
+            return firstKeyframe.frame;
+          }
+        }
```
I'd simplify code, now it's overcautious
```suggestion
        // Handle video rectangle regions - they have sequence with frames
        if (region.type === "videorectangleregion" && region.sequence?.[0]) {
          // Return the first keyframe's frame number
          return region.sequence[0].frame;
        }
```

nass600 committed:
simplified checks

nass600 commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18685147480)

nass600 committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18685147480

nass600 committed:
Merge branch 'develop' into 'fb-utc-325-re'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18685147480
